src/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.cpp

```diff
@@ -49,9 +49,30 @@ Propagator::Propagator(
 )
     : Propagator(aNumericalSolver, aDynamicsArray)
 {
-    // TBM: in the future maybe sanitize the instants of each maneuver to make sure none of them are overalapping
     for (const Maneuver& maneuver : aManeuverArray)
     {
+        // Check if the maneuver's first and last instants overlap with any other maneuver
+        Size duplicateManeuverCounter = 0;
+        for (const Maneuver& otherManeuver : aManeuverArray)
+        {
+            if (maneuver != otherManeuver)  // Skip self-comparison
```

Comment:
```suggestion
            if (maneuver != otherManeuver)
```
The comment makes it more confusing as you aren't skipping in that block.